system/ee/installer/updates/ud_6_02_00.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * This source file is part of the open source project
+ * ExpressionEngine (https://expressionengine.com)
+ *
+ * @link      https://expressionengine.com/
+ * @copyright Copyright (c) 2003-2021, Packet Tide, LLC (https://www.packettide.com)
+ * @license   https://expressionengine.com/license Licensed under Apache License, Version 2.0
+ */
+
+namespace ExpressionEngine\Updater\Version_6_2_0;
+
+/**
+ * Update
+ */
+class Updater
+{
+    public $version_suffix = '';
+
+    /**
+     * Do Update
+     *
+     * @return TRUE
+     */
+    public function do_update()
+    {
+        $steps = new \ProgressIterator([
+            'addMfaColumns',
+            'addMfaMessageTemplate',
+            'dropUnusedMemberColumns',
+        ]);
+
+        foreach ($steps as $k => $v) {
+            $this->$v();
+        }
+
+        return true;
+    }
+
+    private function addMfaColumns()
+    {
+        if (!ee()->db->field_exists('enable_mfa', 'members')) {
+            ee()->smartforge->add_column(
+                'members',
+                [
+                    'enable_mfa' => [
+                        'type' => 'char',
+                        'constraint' => 1,
+                        'default' => 'n',
+                        'null' => false
+                    ]
+                ]
+            );
+        }
+        if (!ee()->db->field_exists('backup_mfa_code', 'members')) {
+            ee()->smartforge->add_column(
+                'members',
+                [
+                    'backup_mfa_code' => [
+                        'type' => 'varchar',
+                        'constraint' => 128,
+                        'default' => null,
+                        'null' => true
+                    ]
+                ]
+            );
+        }
+        if (!ee()->db->field_exists('require_mfa', 'role_settings')) {
+            ee()->smartforge->add_column(
+                'role_settings',
+                [
+                    'require_mfa' => [
+                        'type' => 'char',
+                        'constraint' => 1,
+                        'default' => 'n',
+                        'null' => false
+                    ]
+                ]
+            );
+        }
+        if (!ee()->db->field_exists('mfa_flag', 'sessions')) {
+            ee()->smartforge->add_column(
+                'sessions',
+                [
+                    'mfa_flag' => [
+                        'type' => 'enum',
+                        'constraint' => "'skip','show','required'",
+                        'default' => 'skip',
+                        'null' => false
+                    ]
+                ]
+            );
+        }
+    }
+
+    protected function addMfaMessageTemplate()
+    {
+        $sites = ee('Model')->get('Site')->all();
```

Comment:
I know this is just the site model.. We recently had a conversation about NOT using models in the updater... Should we step away from it here? @intoeetive